sycl/doc/SYCLRuntimeRFC.md

```diff
@@ -0,0 +1,100 @@
+Discourse topic details
+
+- Category: [Runtimes](https://discourse.llvm.org/c/runtimes/64)
+- Title: "RFC: SYCL runtime upstreaming"
+- Tags: sycl
+
+## Overview
+
+SYCL runtime provides an implementation of the SYCL 2020 API specification. It's
+responsible for managing resources for enqueuing tasks to the offload device,
+tracking dependencies between them, and data movement between the host and
+devices. The SYCL runtime is device-agnostic and uses Unified Runtime
+(https://github.com/oneapi-src/unified-runtime) as an external dependency. This
+Unified Runtime serves as an interface layer between the SYCL runtime and
+device-specific backends. Unified Runtime has several adapters that bind to
+various backends in a similar fashion to libomptarget.
+
+This RFC contains a brief overview of the core functionality of the SYCL runtime
+and its major components. The main RFC for SYCL implementation upstreaming can
+be found here:
+https://discourse.llvm.org/t/rfc-add-full-support-for-the-sycl-programming-model/74080
+
+## Host and device code integration
+
+Consider this SYCL code:
+
+```
+#include <sycl/sycl.hpp>
+int main() {
+  sycl::queue q;
+  sycl::range<3> range{8, 8, 8};
+  sycl::buffer<double, 3> b{range};
+  int i = 12;
+  q.submit([&](sycl::handler &cgh) {
+    sycl::accessor acc{b, cgh};
+    cgh.parallel_for(range, [=](sycl::id<3> idx) {
+      acc[idx] += i;
+    });
+  });
+}
+```
+
+The lambda passed to the `sycl::handler::parallel_for` function represents a
+device kernel. When a SYCL application is started or a dynamic SYCL library is
+loaded, all device images contained in the multi-targeted binary are registered
+in the program manager, which is one of the larger components of the SYCL
+runtime, using SYCL's `__tgt_register_lib`
+(https://clang.llvm.org/docs/OffloadingDesign.html#global-constructor-and-destructor)
+counterpart, `__sycl_register_lib`. The device image wrappers generated by SYCL
```

Comment:
This reads a little odd to me. I first read it as stating that `__tgt_register_lib` is part of SYCL because of the distance separation from "counterpart" introduced by the link. Also, `__sycl_register_lib` isn't mentioned in the SYCL specification, so I'm not sure what this is trying to say.

Reply:
It's just supposed to say that internally we use the same mechanism for device image registration as OpenMP with its '__tgt_register_lib`. Changed the wording to make the link placement less awkward.